include/NaluCommNeighbors.hpp

```diff
@@ -0,0 +1,133 @@
+// ***** TEMPORARY COPY OF STK FILE ********
+// Will be delete after stk is updated with performance improvements in
+// stk_util/parallel/CommBufferV.hpp and stk/parallel/CommNeighbors.[hpp,cpp]
+// ***** TEMPORARY COPY OF STK FILE ********
+//
+// Copyright (c) 2013, Sandia Corporation.
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+// 
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+// 
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+// 
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+// 
+//     * Neither the name of Sandia Corporation nor the names of its
+//       contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+#ifndef nalu_CommNeighbors_hpp
+#define nalu_CommNeighbors_hpp
+
+#include <cstddef>                      // for size_t, ptrdiff_t
+#include <vector>
+#include <stk_util/parallel/Parallel.hpp>  // for ParallelMachine
+#include <NaluCommBufferV.hpp>
+
+//------------------------------------------------------------------------
+
+namespace nalu_stk {
+
+class CommNeighbors {
```

Comment:
This looks suspiciously like yet another boundary exchange implementation.

MPI 3 has a boundary exchange implementation.  Dan Ibanez has another one (that wraps / looks like MPI 3's interface).

Reply:
What I'm saying is, why reimplement?